src/components/Table/hoc/withTableSettings/withTableSettings.tsx

```diff
@@ -72,65 +69,63 @@ export function getColumnStringTitle<Data>(column: TableColumnConfig<Data>) {
 export function getActualItems<I>(
     columns: TableColumnConfig<I>[],
     settings: TableSettingsData,
-): SortableItem[] {
-    const newColumnSettings = columns
-        .filter(
-            ({id}) =>
-                id !== actionsColumnId &&
-                id !== selectionColumnId &&
-                settings.every((setting) => setting.id !== id),
-        )
-        .map((column) => ({
-            id: column.id,
-            isSelected: column.meta?.selectedByDefault !== false,
-        }));
-    return settings
-        .filter(({id}) => columns.some((column) => id === column.id))
-        .concat(newColumnSettings)
-        .map(({id, isSelected}) => {
-            const foundColumn = columns.find((column) => column.id === id);
-            const isProtected = Boolean(foundColumn?.meta?.selectedAlways);
-            return {
-                id,
-                isSelected: isProtected ? true : isSelected,
-                isProtected,
-                title: foundColumn ? getColumnStringTitle(foundColumn) : id,
-            };
-        });
-}
-
-function prepareColumnSetupItems(items: SortableItem[]): TableColumnSetupItem[] {
-    return items.map(({id, title, isSelected, isProtected}) => ({
-        id,
-        title,
-        selected: isSelected,
-        required: isProtected,
-    }));
-}
+): TableColumnSetupItem[] {
+    const getTableColumnSetupItem = (
```

Comment:
Lets make this function pure and move it outside.

Reply:
5ae98fa6b94133835922f119d39f3c7424e47bea